eclipse-collections/src/main/java/org/eclipse/collections/impl/set/mutable/AbstractMultiReaderSet.java

```diff
@@ -0,0 +1,1004 @@
+import org.eclipse.collections.api.block.function.primitive.LongFunction;
+import org.eclipse.collections.api.block.function.primitive.ShortFunction;
+import org.eclipse.collections.api.block.predicate.Predicate;
+import org.eclipse.collections.api.block.predicate.Predicate2;
+import org.eclipse.collections.api.block.procedure.Procedure;
+import org.eclipse.collections.api.collection.primitive.MutableBooleanCollection;
+import org.eclipse.collections.api.collection.primitive.MutableByteCollection;
+import org.eclipse.collections.api.collection.primitive.MutableCharCollection;
+import org.eclipse.collections.api.collection.primitive.MutableDoubleCollection;
+import org.eclipse.collections.api.collection.primitive.MutableFloatCollection;
+import org.eclipse.collections.api.collection.primitive.MutableIntCollection;
+import org.eclipse.collections.api.collection.primitive.MutableLongCollection;
+import org.eclipse.collections.api.collection.primitive.MutableShortCollection;
+import org.eclipse.collections.api.factory.Sets;
+import org.eclipse.collections.api.list.MutableList;
+import org.eclipse.collections.api.map.MutableMap;
+import org.eclipse.collections.api.multimap.set.MutableSetMultimap;
+import org.eclipse.collections.api.ordered.OrderedIterable;
+import org.eclipse.collections.api.partition.set.PartitionMutableSet;
+import org.eclipse.collections.api.set.ImmutableSet;
+import org.eclipse.collections.api.set.MultiReaderSet;
+import org.eclipse.collections.api.set.MutableSet;
+import org.eclipse.collections.api.set.ParallelUnsortedSetIterable;
+import org.eclipse.collections.api.set.SetIterable;
+import org.eclipse.collections.api.set.UnsortedSetIterable;
+import org.eclipse.collections.api.set.primitive.MutableBooleanSet;
+import org.eclipse.collections.api.set.primitive.MutableByteSet;
+import org.eclipse.collections.api.set.primitive.MutableCharSet;
+import org.eclipse.collections.api.set.primitive.MutableDoubleSet;
+import org.eclipse.collections.api.set.primitive.MutableFloatSet;
+import org.eclipse.collections.api.set.primitive.MutableIntSet;
+import org.eclipse.collections.api.set.primitive.MutableLongSet;
+import org.eclipse.collections.api.set.primitive.MutableShortSet;
+import org.eclipse.collections.api.tuple.Pair;
+import org.eclipse.collections.impl.collection.mutable.AbstractMultiReaderMutableCollection;
+import org.eclipse.collections.impl.lazy.parallel.set.MultiReaderParallelUnsortedSetIterable;
+import org.eclipse.collections.impl.utility.LazyIterate;
+
+import static org.eclipse.collections.impl.factory.Iterables.mList;
+
+abstract class AbstractMultiReaderSet<T>
+        extends AbstractMultiReaderMutableCollection<T>
+        implements RandomAccess, MultiReaderSet<T>
+{
+    protected MutableSet<T> delegate;
+
+    protected abstract <K> AbstractMultiReaderSet<K> newSetFrom(MutableSet<K> newDelegate);
+
+    @Override
+    protected MutableSet<T> getDelegate()
+    {
+        return this.delegate;
+    }
```

Comment:
Is it possible to implement without a getDelegate method?

Reply:
yes, I can remove it and just use `this.delegate`, should I?

Reply:
actually the `getDelegate()` method is an abstract method of the `AbstractMultiReaderMutableCollection` superclass, so it is needed.

Reply:
@motlin @donraab I fixed this in the latest push